source/sessions/driver-sessions.rst

```diff
@@ -496,6 +499,16 @@ If a driver decides to run a ``KILLCURSORS`` command on the cursor, it also MAY
 run using the same session ID. See the Exceptions below for when it is permissible to not
 include a session ID in a ``KILLCURSORS`` command.
 
+Sessions and Connections
+========================
+A driver MUST only obtain an implicit session after it successfully checks out a connection.
+This limits the number of implicit sessions to never exceed the maximum connection pool size.
+The motivation for this behavior is to prevent many sessions from being created in a scenario
```

Comment:
I think the motivation stated here could benefit from the details actually included in DRIVERS-1030 (regarding not wanting to have more sessions than there are connections and specifically not hitting the TooManyLogicalSessions error). In theory, we could execute all operations with just one session (if we queue them up), but we don't want to do that either. So, I think we should clarify exactly why we are limiting session count to connection count, what the downside is to having too many sessions or too few, and why we only care about implicit sessions and not explicit ones. (Perhaps some of this can go in the Q&A section)

Reply:
Added essentially what's on the ticket with a bit of phrasing, lmk what you think!